zephyr/Makefile

```diff
@@ -71,15 +70,10 @@ $(GENERIC_TARGETS): $(LIBMICROPYTHON)
 $(CLEAN_TARGETS):  clean
 
 $(GENERIC_TARGETS) $(KCONFIG_TARGETS) $(CLEAN_TARGETS):
-	$(RM) -f outdir/$(OUTDIR_PREFIX)/zephyr.lnk
 	$(MAKE) -f Makefile.zephyr BOARD=$(BOARD) CONF_FILE=$(CONF_FILE) $@
 
-$(LIBMICROPYTHON): $(Z_SYSGEN_H)
-build/genhdr/qstr.i.last: $(Z_SYSGEN_H)
-
-$(Z_SYSGEN_H):
-	rm -f $(LIBMICROPYTHON)
-	-$(MAKE) -f Makefile.zephyr BOARD=$(BOARD) CONF_FILE=$(CONF_FILE)
+$(LIBMICROPYTHON): | $(Z_EXPORTS)
```

Comment:
What does this line do? $(Z_EXPORTS) is made a target by an include directive... so order-only prerequisite or not, I can't really see any way it could ever run after the $(LIBMICROPYTHON) rule.

#include <untested_idea.h> ... it occurs to me we should say what we actually mean and make mpconfigport.h depend on sysgen.h (e.g. the config file may inherit values from the config). Once we do that I think everything else will flow correctly (i.e. when we change the config and the uPy C files rebuild and the library will be relinked).